target/linux/ath79/dts/ar9331_glinet_gl-mifi.dts

```diff
@@ -0,0 +1,152 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "ar9331.dtsi"
+
+/ {
+	compatible = "glinet,gl-mifi", "qca,ar9331";
+	model = "GL.iNet GL-MIFI";
+
+	aliases {
+		serial0 = &uart;
+		label-mac-device = &eth0;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		wlan {
+			label = "gl-mifi:green:wlan";
+			gpios = <&gpio 1 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+		lan {
+			label = "gl-mifi:green:lan";
+			gpios = <&gpio 16 GPIO_ACTIVE_HIGH>;
+		};
+
+		wan {
+			label = "gl-mifi:green:wan";
+			gpios = <&gpio 27 GPIO_ACTIVE_HIGH>;
+		};
+
+		3g {
+			label = "gl-mifi:green:3g";
+			gpios = <&gpio 0 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 11 GPIO_ACTIVE_HIGH>;
+			debounce-interval = <60>;
+		};
+	};
+
+	reg_usb_vbus: reg_usb_vbus {
+		compatible = "regulator-fixed";
+		regulator-name = "usb_vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		gpio = <&gpio 6 GPIO_ACTIVE_LOW>;
+	};
+
+	export {
+		compatible = "gpio-export";
+		#size-cells = <0>;
+
+		minipcie {
+			gpio-export,name = "minipcie";
```

Comment:
In ar71xx this has been set up as "3gcontrol", and with active_high...

Reply:
In my opinion the choice of 3gcontrol as name is rather poor as the gpio actually controls the minipci-e slot vcc where various different modules could be (for example I use LTE module instead of 3g).
Also I double checked that the module gets powered on (= is active) when the signal is low so I'm inclined to believe that ar71xx does not do the right thing here.

How do you want to proceed with this one?

Reply:
I'm a fan of using the ar71xx->ath79 process for improvements, so if you know the truth I'm all in for changing stuff if it was wrong/worse in ar71xx.

Reply:
Consider sending a separate patch for ar71xx as well to fix the active_high vs. active_low there ...